### PR TITLE
LevelDB storer Destroy improvements

### PR DESCRIPTION
**Pull request by bogdan-rosianu**
Improved the destroying of the level db storers

### CODE REVIEW

File: storage/leveldb/leveldbSerial.go [hunk 1]
```diff
@@ -260,15 +260,7 @@ func (s *SerialDB) Close() error {
...
-	_ = s.putBatch()
-	s.cancel()
-
-	db := s.makeDbPointerNilReturningLast()
-	if db != nil {
-		return db.Close()
-	}
-
-	return nil
+	return s.prepareClosingUnprotected()
```
Comment (iulianpascalau) [inline storage/leveldb/leveldbSerial.go:263]: suggestion to rename to `doClose` for 2 reasons:
1. unprotected is usually called in conjunction to having a mutex and this method should not be called outside that mutex locking/unlocking
2. it does not "prepare" the closing, it actually "execute" the closing

File: storage/leveldb/leveldbSerial.go [hunk 2]
```diff
@@ -319,6 +301,20 @@ func (s *SerialDB) DestroyClosed() error {
 	return err
 }
 
+// prepareClosingUnprotected will prepare the internal component for closing the database
+// must be called under mutex protection
+func (s *SerialDB) prepareClosingUnprotected() error {
```
Comment (iulianpascalau) [inline storage/leveldb/leveldbSerial.go:306]: can be done in another PR: extract the common code in the level.go file also

File: storage/leveldb/leveldbSerial.go (hunk 1 above)
Comment (bogdan-rosianu) [reply storage/leveldb/leveldbSerial.go]: done

File: storage/leveldb/leveldbSerial.go (hunk 2 above)
Comment (bogdan-rosianu) [reply storage/leveldb/leveldbSerial.go]: added a todo

**Review by iulianpascalau — APPROVED** [review]

File: storage/leveldb/leveldbSerial.go [hunk 3]
```diff
@@ -319,6 +301,21 @@ func (s *SerialDB) DestroyClosed() error {
 	return err
 }
 
+// doClose will handle the closing of the internal components
+// must be called under mutex protection
+// TODO: re-use this function in leveldb.go as well
+func (s *SerialDB) doClose() error {
+	_ = s.putBatch()
+	s.cancel()
```
[the remark this answers is not in the document]
Comment (bogdan-rosianu) [reply storage/leveldb/leveldbSerial.go:309]: no need for this. `cancel()` is set on constructor and in the entire project we rely on the fact that the constructor will return a good object, with no null values

**Review by sstanculeanu — APPROVED** [review]

**Review by gabi-vuls — APPROVED** [review]
System test passed.

**Pull request merged**